File: IUTBeans/IUTBeans/Solution-2/Condiments/WhiteSugar.cs
namespace IUTBeans.Solution_2.Condiments;

public class WhiteSugar:ICondiment
{
    private ICondiment? _condiment;
    public WhiteSugar(ICondiment? condiment)
    {
        _condiment = condiment;
    }
    public string GetDescription()
    {
        return "with White sugar " + (_condiment?.GetDescription() ?? ".");
    }

    public double GetPrice()
    {
        return 5 + (_condiment?.GetPrice() ?? 0);
    }
}

File: Lab_4/src/Condiment/WhiteSugar.java
package Condiment;

public class WhiteSugar implements ICondiment {

    private ICondiment nextCondiment;

    public WhiteSugar(ICondiment nextCondiment){
        this.nextCondiment = nextCondiment;
    }

    public double getPrice(){
        return 5 + nextCondiment.getPrice();
    }
}
